### PR TITLE
Task/um 692-resource-types

### PR DESCRIPTION
**Pull request by kab163**
MemoryResourceTraits updated with resource types (UNKOWN, HOST, FILE, DEVICE, DEVICE_CONST, UM, PINNED). Plus, memory leak fixed in operation_tests.cpp.

### CODE REVIEW

File: tests/integration/memory_resource_traits_tests.cpp [hunk 1]
```diff
@@ -5,28 +5,134 @@
...
 #include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
 #include "umpire/config.hpp"
 #include "umpire/util/MemoryResourceTraits.hpp"
+#include "umpire/strategy/SizeLimiter.hpp"
 
-TEST(Traits, DDR)
+using umpire::MemoryResourceTraits;
+
+class MemoryResourceTraitTest : public ::testing::Test {
```
Comment (davidbeckingsale) [inline tests/integration/memory_resource_traits_tests.cpp:17]: This class is unnecessary.

Comment (kab163) [reply tests/integration/memory_resource_traits_tests.cpp]: Indeed! Whoopsies

**Comment by kab163** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
Please update the CHANGELOG 😄

**Comment by kab163** [comment]
LGTM yet again

**Comment by kab163** [comment]
LGTM

**Pull request merged**